python/src/opendp/mod.py

```diff
@@ -513,10 +635,35 @@ def __init__(self, value, query_type):
     def __call__(self, query):
         from opendp.core import queryable_eval
         return queryable_eval(self.value, query)
-
+    
     def __repr__(self) -> str:
         return f"Queryable(Q={self.query_type})"
 
+class OdometerQueryable(object):
+    '''
+    Odometer Queryables are used for instances of odometers like :ref:`fully adaptive composition <fully-adaptive-composition>`.
+
+    Can be created via :py:func:`make_fully_adaptive_composition <opendp.combinators.make_fully_adaptive_composition>`.
+    '''
+    def __init__(self, value, invoke_type):
+        self.value = value
+        self.invoke_type = invoke_type
```

Comment:
Looking through the backlog, I think you proposed checking that the value is a pointer, to prevent the misuse of constructors.

- #2261 

Could you try that out here to confirm that it works, and then that pattern could be applied more broadly?